lib/comment.lib.js

```diff
@@ -0,0 +1,138 @@
+const models = require('../models/index.model');
+
+const QuestionComments = models.question_comments;
+const AnswerComments = models.answer_comments;
+
+// Comments on question related
+const fetchCommentOnQuestion = async (value, raw = false) => QuestionComments.findOne(
+  {
+    where: { ...value }, raw,
+  },
+);
+
+const fetchCommentsOnQuestion = async (question_id, raw = false) => {
+  try {
+    const comments = await QuestionComments.findAll({
+      where: { question_id },
+      raw,
+    });
+    return comments;
+  } catch (error) {
+    throw new Error(`${error.message}`);
+  }
+};
+
+const commentOnQuestion = async (commentDetails) => {
+  const {
+    user_id, question_id, comment_text,
+  } = commentDetails;
+  try {
+    const comment = await QuestionComments.create({ user_id, question_id, comment_text });
+    return comment;
+  } catch (error) {
+    throw new Error(`${error.message}`);
+  }
+};
+
+const editCommentOnQuestion = async (commentDetails) => {
+  const {
+    user_id, id, comment_text,
+  } = commentDetails;
+  try {
+    const comment = await fetchCommentOnQuestion({ user_id, id });
+    if (comment) {
+      comment.comment_text = comment_text;
+      await comment.save();
+      return comment;
+    }
```

Comment:
Check if it's the user that made that comment is the one editing.

Reply:
On line 42 its actually checking if its the logged-in user that made the comment

Reply:
Yeah, thanks, I overlooked the `user_id` passed to the query